sdk/storage/storage-file/src/DirectoryClient.ts

```diff
@@ -722,12 +750,151 @@ export class DirectoryClient extends StorageClient {
     });
   }
 
+  /**
+   * Returns an AsyncIterableIterator for DirectoryListHandlesResponse
+   *
+   * @private
+   * @param {string} [marker] A string value that identifies the portion of the list to be
+   *                          returned with the next list handles operation. The operation returns a
+   *                          marker value within the response body if the list returned was not complete.
+   *                          The marker value may then be used in a subsequent call to request the next
+   *                          set of list items.
+   * @param {DirectoryListHandlesSegmentOptions} [options] Options to list handles operation.
+   * @returns {AsyncIterableIterator<Models.DirectoryListHandlesResponse>}
+   * @memberof DirectoryClient
+   */
+  private async *iterateHandleSegments(
+    marker?: string,
+    options: DirectoryListHandlesSegmentOptions = {}
+  ): AsyncIterableIterator<Models.DirectoryListHandlesResponse> {
+    let listHandlesResponse;
+    if (!!marker || marker === undefined) {
```

Comment:
@bterlson As you suggested this seems working to solve the continuation token issue.

Reply:
This should be done for the other listing methods, right? Separate PR?

Reply:
Yup all the paged async iterators. If this is good. Fix for othesrs could be in a seperate PR for storage, and another one for master branch.